OpenID_Connect/oidc_config.sh

```diff
@@ -37,6 +38,123 @@ function print_usage() {
   exit 1
 }
 
+function get_settings_id() {
+  SETTINGS_ID=`curl $CURL_OPTS \
+    -H "Authorization: Bearer $API_TOKEN" \
+    -X GET \
+    ${SETTINGS_ENDPOINT} | jq '.authenticationSettings | .[] | select(.type=="'"${SSO_KEYWORD}"'") | .id'`
+}
+
+function exit_if_no_settings_id() {
+  if [ -z "$SETTINGS_ID" ] ; then
+    echo "No ${SSO_KEYWORD} settings are set"
+    echo
+    exit 0
+  fi
+}
+
+function get_active_settings_type() {
+  ACTIVE_SETTINGS_TYPE=`curl $CURL_OPTS \
+    -H "Authorization: Bearer $API_TOKEN" \
+    -X GET \
+    $ACTIVE_ENDPOINT | jq '.activeSettings | .type'`
+}
+
+function set_provider_variables() {
+  if [ -z "$ISSUER_URL" -o -z "$CLIENT_ID" -o -z "$CLIENT_SECRET" ] ; then
+    echo "To change settings, you must enter values for Issuer URL, Client ID, and Client Secret"
+    echo
+    print_usage
+  fi
+}
+
+function get_settings_version() {
+  VERSION=`curl $CURL_OPTS \
+    -H "Authorization: Bearer $API_TOKEN" \
+    -X GET \
+    ${SETTINGS_ENDPOINT} | jq '.authenticationSettings | .[] | select(.type=="'"${SSO_KEYWORD}"'") | .version'`
+}
+
+function set_as_active_setting() {
+  get_settings_id
+  curl $CURL_OPTS \
+    -H "Authorization: Bearer $API_TOKEN" \
+    -X PUT \
+    $ACTIVE_ENDPOINT/$SETTINGS_ID | ${JSON_FILTER}
+}
+
+function set_settings() {
+  set_provider_variables
+  get_settings_id
+  if [ -z "$SETTINGS_ID" ] ; then
+    curl $CURL_OPTS \
+      -H "Authorization: Bearer $API_TOKEN" \
+      -H "Content-Type: application/json" \
+      -X POST \
+      -d '{
+        "authenticationSettings": {
+          "type": "'"${SSO_KEYWORD}"'",
+          "settings": {
+            "issuer":"'"${ISSUER_URL}"'",
+            "clientId":"'"${CLIENT_ID}"'",
+            "clientSecret":"'"${CLIENT_SECRET}"'",
+            "metadataDiscovery":true}}}' \
```

Comment:
What if the user wants to disable `metadataDiscovery` and provide the metadata explicitly?

Reply:
Same here:

> This was originally set at the script. I didn't wanted to change anything in how the scripts where invoked so DOCS don't have to run for the next onprem. We can have a task after the release to improve and doc this without this rush. What do you think?

Reply:
Same here. Ok with your change as it keeps the logic the same.